File: src/main/java/com/bazbatlabs/bricksmash/views/MenuView.java
package com.bazbatlabs.bricksmash.views;

import com.bazbatlabs.bricksmash.models.*;

public final class MenuView {

    private static final int CYCLE_LENGTH = 200;
    private static final float CYCLE_COEFFICIENT = (float)(2 * Math.PI) / (float)CYCLE_LENGTH;

    private final Menu menu;

    private final ImageMap images;
    private final SoundMap sounds;
    private final Artist artist;

    private float r;
    private float g;
    private float b;

    private int counter;

    public MenuView(Menu menu, ImageMap images, SoundMap sounds, Artist artist) {

        this.menu = menu;

        this.images = images;
        this.sounds = sounds;
        this.artist = artist;

        this.r = 1.0f;
        this.g = 1.0f;
        this.b = 1.0f;

        this.counter = 0;
    }

    public void draw() {
        artist.startDrawing();

        float radians = CYCLE_COEFFICIENT * (counter % CYCLE_LENGTH);

        b = r = (float)Math.abs(Math.sin(radians));
        g = (float)Math.abs(Math.cos(radians));

        Color color = new Color(r, g, b, 1f);

        artist.drawImage(new Vec2(100f, 200f), new Vec2(200f, 20f), images.get("TITLE"), color);

        artist.finishDrawing();

        counter++;
    }
}

File: src/main/scala/views/MenuView.scala
package com.bazbatlabs.bricksmash.views

import com.bazbatlabs.bricksmash.models._
import com.bazbatlabs.bricksmash.lib._

class MenuView(val menu: Menu, val images: ImageMap, val sounds: SoundMap, val artist: Artist) {

  val CycleLength = 200
  val CycleCoefficient =
    (2 * Math.PI).asInstanceOf[Float] / CycleLength.asInstanceOf[Float]

  var r = 1f
  var g = 1f
  var b = 1f
  var counter = 0

  def draw() {
    artist.startDrawing()

    val radians = CycleCoefficient * (counter % CycleLength)
    r = Math.abs(Math.sin(radians)).asInstanceOf[Float]
    b = r
    g = Math.abs(Math.cos(radians)).asInstanceOf[Float]
    val color = Color(r, g, b, 1f)

    val titleImage = images.get("TITLE")
    val dimensions = artist.dimensions

    val titleHeight = dimensions.y / 12f
    val titleWidth = titleHeight * (titleImage.w / titleImage.h)

    val titleSize = Vec2(titleWidth, titleHeight)

    val x = (dimensions.x - titleSize.x) / 2f
    val y = (dimensions.y - titleSize.y) / 2f

    artist.drawImage(Rect(Vec2(x, y), titleSize), titleImage, color)
    artist.finishDrawing()

    counter += 1
  }
}
